.github/workflows/tests.yml

```diff
@@ -40,6 +45,93 @@ jobs:
       - name: Run Unit Test
         run: LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH make gosdk-test
 
+  system-test:
+    needs: go-test
+    runs-on: [ tests-suite ]
+    steps:
+      - name: "Get current PR"
+        uses: jwalton/gh-find-current-pr@v1
+        id: findPr
+        with:
+          github-token: ${{ github.token }}
+
+      - name: "Set PR Status Pending"
+        uses: niteoweb/pull_request_status_action@v1.0.0
+        if: steps.findPr.outputs.number
+        with:
+          pr_number: ${{ steps.findPr.outputs.pr }}
+          description: "System tests running with default config..."
+          state: "pending"
+          repository: ${{ github.repository }}
+          context: "0Chain System Tests"
+          target_url: https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+
+      - name: "Setup"
+        run: |
+          echo "CURRENT_BRANCH_HEAD=${{ github.event.pull_request.head.sha }}" >> $GITHUB_ENV
+          echo "NETWORK_URL=$(echo dev-${RUNNER_NAME:(-1)}.devnet-0chain.net)" >> $GITHUB_ENV
+          echo "RUNNER_NUMBER=${RUNNER_NAME:(-1)}" >> $GITHUB_ENV
+
+      - name: "Deploy 0Chain"
+        uses: 0chain/actions/deploy-0chain@master
+        with:
+          kube_config: ${{ secrets[format('DEV{0}KC', env.RUNNER_NUMBER)] }}
+          teardown_condition: "TESTS_PASSED"
+          miner_image:  staging
+          sharder_image: staging
+          blobber_image: staging
+          validator_image: staging
+          zbox_image: staging
+          zblock_image: staging
+          zdns_image: staging
+          explorer_image: latest
+          zproxy_image: staging
+          zsearch_image: staging
+          blobber_stake_image: latest
+
+      - name: "Run System tests"
+        uses: 0chain/actions/run-system-tests@master
+        with:
+          custom_go_sdk_version: ${{ env.CURRENT_BRANCH_HEAD }}
```

Comment:
This will 
1. checkout zwalletcli and update it to use the gosdk version from the current branch
2. checkout zboxcli and update it to use the gosdk version from the current branch
3. run system tests 

If tests fail, or if staging CLI is not compatible with current SDK then manual system tests can be ran as in other repos